x-pack/plugins/cases/server/client/cases/utils.ts

```diff
@@ -236,16 +240,47 @@ export const createIncident = async ({
   return { incident, comments };
 };
 
-export const getEntity = (entity: EntityInformation): string =>
-  (entity.updatedBy != null
-    ? entity.updatedBy.full_name
-      ? entity.updatedBy.full_name
-      : entity.updatedBy.username
-    : entity.createdBy != null
-    ? entity.createdBy.full_name
-      ? entity.createdBy.full_name
-      : entity.createdBy.username
-    : '') ?? '';
+export const getEntity = (
+  entity: EntityInformation,
+  userProfiles?: Map<string, UserProfile>
+): string => {
+  return (
+    getDisplayName(entity.updatedBy, userProfiles) ??
+    getDisplayName(entity.createdBy, userProfiles) ??
+    ''
+  );
+};
+
+const getDisplayName = (
+  user: User | null | undefined,
+  userProfiles?: Map<string, UserProfile>
+): string | undefined => {
+  if (user == null) {
+    return;
+  }
+
+  if (user.profile_uid != null) {
+    const updatedByProfile = userProfiles?.get(user.profile_uid);
+
+    if (updatedByProfile != null) {
+      return (
+        validOrUndefined(updatedByProfile.user.full_name) ??
+        validOrUndefined(updatedByProfile.user.username)
+      );
+    }
+  }
+
+  // TODO: if username isn't defined should we fallback to the next field (aka createdBy)? That's different than how it was before
+  return validOrUndefined(user.full_name) ?? validOrUndefined(user.username) ?? '';
```

Comment:
This would be diversion from the previous behavior but if `user.username` is undefined, we could return undefined instead so it would fallback to the next field (aka if we were looking at `updatedBy` it would look at `createdBy` next and then default to an empty string).

Reply:
I think it is best to leave it as it was because the `updatedBy` user can be different from the `createdBy` user.